ompi/proc/proc.h

```diff
@@ -369,16 +369,42 @@ static inline bool ompi_proc_is_sentinel (ompi_proc_t *proc)
     return (intptr_t) proc & 0x1;
 }
 
-static inline intptr_t ompi_proc_name_to_sentinel (opal_process_name_t name)
+/*
+ * we assume an ompi_proc_t is at least aligned on two bytes,
+ * so if the LSB of a pointer to an ompi_proc_t is 1, we have to handle
+ * this pointer as a sentinel instead of a pointer.
+ * a sentinel can be seen as a 64 bits array with the following format :
+ * - bit  0     : 1
+ * - bits 1-15  : local jobid
+ * - bits 16-31 : job family
+ * - bits 32-63 : vpid
```

Comment:
Sentinel bit is the LSB, aka bit zero.
see my comment in the code above
an ompi_proc_t* can either be a genuine pointer, or a sentinel.
that would require 65 bits.
the trick is pointer are always aligned on two bytes, so the LSB of a pointer is always zero, and we can use this bit as the sentinel bit.
if it is one, it means this is not a pointer but a sentinel.
that still means we have to "pack" the opal_process_name_t into the remaining 63 bits, and we decided to drop the MSB of the local jobid.

note that cannot work "as is" 32 bits.

my best bet is to simply disable cutoff at runtime on 32 bits arch

an other option to keep memory usage low is to use the sentinel bit as an opal_process_name_t* instead of a "packed" opal_process_name_t, but I am not sure it is worth it.

I will likely use jobid and family as uint32_t as you mentionned, and fix the abstraction 

makes more sense now ?